test/integration/uds_integration_test.h

```diff
@@ -46,4 +47,36 @@ class UdsIntegrationTest
 protected:
   const bool abstract_namespace_;
 };
+
+class UdsListenerIntegrationTest : public HttpIntegrationTest,
+                                   public testing::TestWithParam<std::tuple<bool>> {
+public:
+  UdsListenerIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, Network::Address::IpVersion::v4),
+        abstract_namespace_(std::get<0>(GetParam())) {}
+
+  void initialize() override;
+
+  std::string getAdminSocketName() {
+    const std::string path = TestEnvironment::unixDomainSocketPath("admin.sock");
+    if (!abstract_namespace_) {
+      return path;
+    }
+    return "@" + path;
+  }
+
+  std::string getListenerSocketName() {
+    const std::string path = TestEnvironment::unixDomainSocketPath("listener_0.sock");
+    if (!abstract_namespace_) {
+      return path;
+    }
+    return "@" + path;
```

Comment:
This and the above method could be slightly refactored to reduce common code.

Reply:
Done.